src/llmq/quorums.cpp

```diff
@@ -517,10 +528,12 @@ size_t CQuorumManager::GetQuorumRecoveryStartOffset(const CQuorumCPtr pQuorum, c
     });
     std::sort(vecProTxHashes.begin(), vecProTxHashes.end());
     size_t nIndex{0};
-    for (size_t i = 0; i < vecProTxHashes.size(); ++i) {
-        if (activeMasternodeInfo.proTxHash == vecProTxHashes[i]) {
-            nIndex = i;
-            break;
+    {
+        for (size_t i = 0; i < vecProTxHashes.size(); ++i) {
+            if (WITH_LOCK(activeMasternodeInfoCs, return activeMasternodeInfo.proTxHash == vecProTxHashes[i])) {
```

Comment:
Should we just lock outside of the loop here instead? And if we don't, then should remove redundant scope outside of loop

Reply:
We should just lock outside, fixed it.